lib/src/code_generator/utils.dart

```diff
@@ -70,3 +70,73 @@ String makeDoc(String text) {
 
   return s.toString();
 }
+
+class YamlWriter {
```

Comment:
package:yaml_edit can write yaml files.

```dart
String _toYamlString(Object yamlEncoding) => (YamlEditor('')
      ..update(
          [],
          wrapAsYamlNode(
            yamlEncoding,
            collectionStyle: CollectionStyle.BLOCK,
            scalarStyle: ScalarStyle.DOUBLE_QUOTED,
          )))
    .toString();
```

If this file becomes to burdensome to maintain, we could depend on that package. (The Dart SDK itself already depends on it internally.)

Reply:
Oh cool, I've replaced my small implementation with it.